### ISSUE TITLE
service instances health state not updated when host is removed

### ISSUE THREAD

**Issue by cjellick** [rancher/rancher#3763]
This is tough to reproduce, but here is the general idea:
An instance had healthchecks from two hosts and one of the two hosts was removed. At the same time, the instance's state went from healthy to unhealthy. But, the instance was not updated to unhealthy because the event came in right before the host was removed and at that time, that host was still reporting the instance as healthy. So, because we had one healthy and one healthy report, we left the instance as healthy.

Solution is to recalculate an instance's health state upon host removal.

@sangeethah ran into this when testing VM HA. She stopped a host which a host, which caused a lot of healthchecks events to fire at the same time.

**Comment by alena1108** [comment]
@sangeethah not easy to reproduce, so could you just keep an eye on it while doing storage testing?

**Comment by sangeethah** [comment]
Tested with server version - V0.63.0 ( and v0.63.0-rc3)

There was one issue that regressed because of this fix - https://github.com/rancher/rancher/issues/3933.
This issue is addressed in V0.63.0 .

Tested generic use cases relating to multiple host down scenarios and HA scenarios work as expected: 
Following 2 issues were found which are not related to this issue - #3959 and #3961.

Performed the following test to verify this fix:

Updated state of instances as "Unhealthy" in healthcheck_instance_host_map for 2 of the hosts.

VM continues to remain in "Unhealthy" state which is as expected.

Power down the host that still has the instance health marked "healthy" for this instance.

Once this host goes to "reconnecting" state , Instance gets marked "Unhealthy" and gets recreated in a different host as expected.

```
mysql> select * from healthcheck_instance_host_map where instance_id=18;
+----+------+------------+----------------------------+--------------------------------------+-------------+---------+---------------------+---------------------+---------------------+------+-------------------------+---------+--------------------+--------------+-------------+
| id | name | account_id | kind                       | uuid                                 | description | state   | created             | removed             | remove_time         | data | healthcheck_instance_id | host_id | external_timestamp | health_state | instance_id |
+----+------+------------+----------------------------+--------------------------------------+-------------+---------+---------------------+---------------------+---------------------+------+-------------------------+---------+--------------------+--------------+-------------+
| 29 | NULL |          5 | healthcheckInstanceHostMap | fd0e6d25-f24f-4f79-afbe-3beb00bd7f51 | NULL        | removed | 2016-03-14 18:35:41 | 2016-03-14 18:38:24 | 2016-03-14 18:39:24 | {}   |                      12 |       1 |         1457982100 | healthy      |          18 |
| 33 | NULL |          5 | healthcheckInstanceHostMap | 638b1131-b43b-4be4-9240-bd5315b8be83 | NULL        | active  | 2016-03-14 19:01:10 | NULL                | NULL                | {}   |                      12 |       1 |         1457982100 | unhealthy    |          18 |
| 35 | NULL |          5 | healthcheckInstanceHostMap | 41dd64ed-5329-4956-8cdb-aaf98ec94769 | NULL        | active  | 2016-03-14 19:01:16 | NULL                | NULL                | {}   |                      12 |       2 |         1457982097 | unhealthy    |          18 |
| 41 | NULL |          5 | healthcheckInstanceHostMap | 5c16f42d-fcb2-47f5-98f4-eb62f528c071 | NULL        | active  | 2016-03-14 19:08:42 | NULL                | NULL                | {}   |                      12 |       4 |         1457982571 | healthy      |          18 |
+----+------+------------+----------------------------+--------------------------------------+-------------+---------+---------------------+---------------------+---------------------+------+-------------------------+---------+--------------------+--------------+-------------+
```
